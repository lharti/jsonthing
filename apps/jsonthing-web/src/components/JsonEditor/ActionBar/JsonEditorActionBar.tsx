import { Button } from '@/components/ui/Button'
import { prettifyJson } from '@/lib/prettify-json'
import React from 'react'

export interface JsonEditorActionBarProps {
    className?: string

    editorContent: string
    setEditorContent: (value: string) => void
}

export const JsonEditorActionBar: React.FC<JsonEditorActionBarProps> = ({
    className,

    editorContent,
    setEditorContent,
}) => {
    return (
        <div className={className}>
            <Button
                variant="outline"
                className="my-2 h-6 rounded-none bg-transparent p-0 px-2"
                onClick={() => {
                    const prettyEditorContent = prettifyJson(editorContent)

                    setEditorContent(prettyEditorContent)
                }}
            >
                {'Prettify'}
            </Button>
        </div>
    )
}
