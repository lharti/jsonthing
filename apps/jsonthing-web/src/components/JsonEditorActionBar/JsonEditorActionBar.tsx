import { useUpdateDoc } from '@/hooks/useUpdateDoc'
import { prettifyJson } from '@/lib/prettify-json'
import { cn } from '@/lib/utils'
import {
    IconClipboardCopy,
    IconDeviceFloppy,
    IconWand,
} from '@tabler/icons-react'
import { useParams } from 'next/navigation'
import React from 'react'
import { JsonEditorActionBarBtn } from './ActionBarBtn'

export interface JsonEditorActionBarProps {
    className?: string

    editorContent: string
    setEditorContent: React.Dispatch<React.SetStateAction<string>>
}

export const JsonEditorActionBar: React.FC<JsonEditorActionBarProps> = ({
    className,

    editorContent,
    setEditorContent,
}) => {
    const prettifyContent = () => {
        setEditorContent(prettifyJson(editorContent))
    }

    const copyToClipboard = () => {
        navigator.clipboard.writeText(editorContent)
    }

    const { id } = useParams()

    const { updateDoc } = useUpdateDoc()

    const save = () => {
        updateDoc({
            id: id as string,

            payload: {
                content: editorContent,
            },
        })
    }

    return (
        <div className={cn('flex', className)}>
            <JsonEditorActionBarBtn
                iconOnly
                label="Prettify"
                Icon={IconWand}
                onClick={() => prettifyContent()}
            />

            <JsonEditorActionBarBtn
                iconOnly
                label="Copy"
                Icon={IconClipboardCopy}
                onClick={() => copyToClipboard()}
            />
            <JsonEditorActionBarBtn
                variant="outline"
                label="Save"
                Icon={IconDeviceFloppy}
                onClick={() => save()}
            />
        </div>
    )
}
