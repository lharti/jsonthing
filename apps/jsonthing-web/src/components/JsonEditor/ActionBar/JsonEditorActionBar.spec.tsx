import { render, screen } from '@testing-library/react'
import React from 'react'
import { JsonEditorActionBar } from './JsonEditorActionBar'

describe('<JsonEditorActionBar />', () => {
    it('should render', () => {
        expect.assertions(1)

        const { container } = render(
            <JsonEditorActionBar
                editorContent="{}"
                setEditorContent={jest.fn()}
            />,
        )

        expect(container).toMatchSnapshot('<JsonEditorActionBar />')
    })

    it('should prettify editor content', () => {
        expect.assertions(1)

        const setEditorContent = jest.fn(value => value)

        const uglyJson = `
            {"a":1,"b":2,
            "c": 3}`

        render(
            <JsonEditorActionBar
                editorContent={uglyJson}
                setEditorContent={setEditorContent}
            />,
        )

        screen.getByText('Prettify').click()

        const newEditorContent = setEditorContent.mock.results[0].value

        expect(newEditorContent).toMatchInlineSnapshot(`
            "{
              "a": 1,
              "b": 2,
              "c": 3
            }"
        `)
    })
})
