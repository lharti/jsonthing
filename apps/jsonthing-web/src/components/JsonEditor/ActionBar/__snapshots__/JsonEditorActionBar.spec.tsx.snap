// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JsonEditorActionBar /> should render: <JsonEditorActionBar /> 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors dark:focus-visible:ring-gray-300 disabled:pointer-events-none disabled:opacity-50 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-gray-950 border border-gray-200 shadow-sm dark:border-gray-800 dark:bg-gray-950 dark:hover:bg-gray-800 dark:hover:text-gray-50 hover:bg-gray-100 hover:text-gray-900 my-2 h-6 rounded-none bg-transparent p-0 px-2"
    >
      Prettify
    </button>
  </div>
</div>
`;
